bpf/lib/policy.h

```diff
@@ -186,7 +186,28 @@ __policy_can_access(const void *map, struct __ctx_buff *ctx, __u32 local_id,
 				return DROP_POLICY_DENY;
 			return policy->proxy_port;
 		}
+		
+		/* L4-only lookup, If not fill out port, podSelector and namespaceSelector */
+		key.dport = 0;
+		policy = map_lookup_elem(map, &key);
+		if (likely(policy)) {
+			account(ctx, policy);
+			*match_type = POLICY_MATCH_L4_ONLY;
+			if (unlikely(policy->deny))
+				return DROP_POLICY_DENY;
+			return policy->proxy_port;
+		}
+		
+		/* L4-only lookup, If not fill out port ,but use podSelector and namespaceSelector */
 		key.sec_label = remote_id;
+		policy = map_lookup_elem(map, &key);
+		if (likely(policy)) {
+			account(ctx, policy);
+			*match_type = POLICY_MATCH_L4_ONLY;
+			if (unlikely(policy->deny))
+				return DROP_POLICY_DENY;
+			return policy->proxy_port;
+		}
```

Comment:
@xiake468 Do you have specific use cases in mind where particular kinds of applications would benefit from having policies that look like this, vs. instead just having pure L3 policies? I think that these use cases could help to inform the balance of introducing new logic like this.

Reply:
One mitigation for the performance concern would be to identify such policies when they are inserted into the agent, then conditionally enable this new logic.

Reply:
I think now we can only resolve the case that use podselector or namespaceselector. As for external ip , because it affects performance and it is not security. So we can consider it later whether to support.